crates/app/src/gas/storage.rs

```diff
@@ -58,6 +58,21 @@ where
     fn load_with_gas(&self, storage: &dyn Storage, gas_tracker: GasTracker, key: K)
         -> StdResult<T>;
 
+    fn has_with_gas(
+        &self,
+        storage: &dyn Storage,
+        gas_tracker: GasTracker,
+        key: K,
+    ) -> StdResult<bool>;
+
+    fn save_with_gas(
+        &self,
+        storage: &mut dyn Storage,
+        gas_tracker: GasTracker,
+        key: K,
+        value: &T,
+    ) -> StdResult<()>;
```

Comment:
Minor styling - let's put `save_with_gas` below `range_with_gas`, because the way I order these is all non-mutable methods first, then all mutable ones.